cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -164,7 +164,8 @@
     this.programStateWriter = programStateWriter;
     this.capabilityReader = capabilityReader;
     this.artifactRepository = artifactRepository;
-    this.runRecordCounter = new RunRecordCounter(runtimeService);
+    this.runRecordCounterService = new RunRecordCounterService(cConf, runtimeService);
+    runRecordCounterService.startAndWait();
```

Comment:
Generally one service should not manage lifecycle of other.

Reply:
Fixed. Was trying to quickly test the service so did it this way.